src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -306,10 +394,8 @@ ZoneSet::MoveWindowIntoZoneByPoint(HWND window, HWND zoneWindow, POINT ptClient)
         zoneDrop->RemoveWindowFromZone(window, !IsZoomed(window));
     }
 
-    if (auto zone = ZoneFromPoint(ptClient))
-    {
-        zone->AddWindowToZone(window, zoneWindow, true);
```

Comment:
the third argument is `true`, this needs to be passed to `MoveWindowIntoZoneByIndexSet` that in turn will pass it to `MoveWindowIntoZoneByIndex`  and finally to `AddWindowToZone` that currently is called like this `zone->AddWindowToZone(window, windowZone, false);` passing always `false` as third argument.

Let me know if I didn't explain myself.

Reply:
Forgot to mention that this is in order to execute Zone::StampZone() that will set the zone index as a prop on the window.

Reply:
I understand what the problem is and I'm fixing it right now. Basically now the feature "During zone layout changes, windows assigned to a zone will match new size/positions" does not work.

Reply:
Correct.

Reply:
Why not just change the third argument `zone->AddWindowToZone(window, windowZone, false);` to `true`, stamping a window won't do any harm.

Reply:
`MoveWindowsOnDisplayChange` calls GetProp to figure out if a window is in a zone and to get the index.

Reply:
I pushed a commit to fix this and checked that windows stay in their zones when the layout or the screen resolution changes.

Reply:
Tested, it work as expected for the case of a window snapped to a single zone.
For a window snapped to two zone it doesn't move the window. I guess that for now we can accept this limitation. Also a window snapped to two zones, after being closed and re-launched will be zoned to one zone. To change this one we will have to change the schema of the json. Let's consider it for a future improvement.